### PR TITLE
[polymetis] Fix timestamp representation

### PR DESCRIPTION
**Pull request by exhaustin**
# Description

Naively converting the timestamp to a float would result in loss of precision, causing time differences to be incorrectly calculated.

- Change the tensor representation of a timestamp to a int32 tensor of size 2, with the first element representing seconds and the second representing nanoseconds.
- Add helper functions to easily obtain time differences between two timestamps.

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Before this change, timestamps would get converted to the same float value due to [loss of significance](https://en.wikipedia.org/wiki/Loss_of_significance), and thus trying to get differences between timestamps would always result in 0 seconds.

After the change, helper functions return accurate time differences between two timestamps.

# Testing

Add integration test `test/scripts/3_timestamps.py`

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: polymetis/polymetis/python/torchcontrol/utils/time_utils.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import torch
```
Comment (1heart) [inline polymetis/polymetis/python/torchcontrol/utils/time_utils.py:1]:
```suggestion
# Copyright (c) Facebook, Inc. and its affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.
import torch
```

File: polymetis/tests/scripts/3_timestamps.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        self.i = 0
+        self.ts_prev = torch.zeros(2).to(torch.int32)
+
+    def forward(self, state_dict: Dict[str, torch.Tensor]):
+        ts = state_dict["timestamp"]
+
+        # Check timestamp
+        if self.i > self.warmup_steps:
+            t_diff = toco.utils.timestamp_diff_seconds(ts, self.ts_prev)
+            assert torch.allclose(t_diff, self.dt, atol=1e-3)  # millisecond accuracy
```
Comment (1heart) [inline polymetis/tests/scripts/3_timestamps.py:38]: Is this test dependent on the machine it's run on? E.g. do we expect this test to become flaky if CircleCI's VMs get slow for whatever reason?

**Review by 1heart — APPROVED** [review]
Minor nits; otherwise looks good, thanks! This was harder than I expected timestamps to be :D

File: polymetis/tests/scripts/3_timestamps.py (hunk 2 above)
Comment (exhaustin) [reply polymetis/tests/scripts/3_timestamps.py:38]: Yes which is why I'm setting a very large tolerance.
An alternative is that we can also test if t_diff > 0, but the current test would also ensure that the frequency provided in the metadata from the simulation client is correct so I went for it.
I'm fine with either, thoughts?

Comment (1heart) [reply polymetis/tests/scripts/3_timestamps.py:38]: sgtm
